Framework/PythonInterface/plugins/algorithms/HB2AReduce.py

```diff
@@ -118,14 +118,15 @@ def validateInputs(self):
             if ((ipts == Property.EMPTY_INT) or len(self.getProperty("ScanNumbers").value) == 0):
                 issues["Filename"] = 'Must specify either Filename or IPTS AND ScanNumbers'
 
+            directory = f'/HFIR/HB2A/IPTS-{ipts}'
             if self.getProperty("Exp").value == Property.EMPTY_INT:
-                exp_list = sorted(e for e in os.listdir('/HFIR/HB2A/IPTS-{0}'.format(ipts))
-                                  if 'exp' in e)
-                if len(exp_list) > 1:
-                    exps = ','.join(e.replace('exp', '') for e in exp_list)
-                    issues[
-                        "Exp"] = 'Multiple experiments found in IPTS-{}. You must set Exp to one of {}'.format(
-                            ipts, exps)
+                if os.path.isdir(directory):
+                    exp_list = sorted(e for e in os.listdir(directory) if 'exp' in e)
+                    if len(exp_list) > 1:
+                        exps = ','.join(e.replace('exp', '') for e in exp_list)
+                        issues["Exp"] = f'Multiple experiments found in IPTS-{ipts}. You must set Exp to one of {exps}'
+            else:
```

Comment:
Good spot, this should hopefully be fixed now

Reply:
The code looks good now but this is still crashing. It's crashing in the algorithm, though, rather than the dialog, so to some extent is not related to this PR. However, I'm confused why the new checks aren't stopping it being run from the dialog!